recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -34,15 +41,26 @@ this.RecipeRunner = {
     const durabilityManager = new SandboxManager();
     Storage.seedDurability(durabilityManager.sandbox);
 
-    let delay;
     if (prefs.getBoolPref("dev_mode")) {
-      delay = 0;
+      // Run right now in dev mode
+      this.run();
+    }
+
+    this.updateRunInterval();
+    if (timerManager.unregisterTimer) {
+      CleanupManager.addCleanupHandler(() => timerManager.unregisterTimer(TIMER_NAME));
     } else {
-      // startup delay is in seconds
-      delay = prefs.getIntPref("startup_delay_seconds") * 1000;
+      // Since we can't unregister the timer yet (Bug 1350471), re-register
+      // with a period so long as to never actually run
```

Comment:
Yesterday we decided not to ship to 52, which was previously a problem with not handling this, so I think this is safe to remove now.